src/Language/PureScript/TypeChecker/Types.hs

```diff
@@ -838,18 +838,19 @@ checkProperties expr ps row lax = convert <$> go ps (toRowPair <$> ts') r' where
   go [] ((p, _): _) _ | lax = return []
                       | otherwise = throwError . errorMessage $ PropertyIsMissing p
   go ((p,_):_) [] (REmptyKinded _ _) = throwError . errorMessage $ AdditionalProperty $ Label p
-  go ((p,v):ps') ts r =
-    case lookup (Label p) ts of
-      Nothing -> do
-        v'@(TypedValue' _ _ ty) <- infer v
-        rest <- freshTypeWithKind (kindRow kindType)
-        unifyTypes r (srcRCons (Label p) ty rest)
-        ps'' <- go ps' ts rest
-        return $ (p, v') : ps''
-      Just ty -> do
-        v' <- check v ty
-        ps'' <- go ps' (delete (Label p, ty) ts) r
-        return $ (p, v') : ps''
+  go ((p,v):ps') ts r = 
+    withErrorMessageHint (ErrorInRowLabel (Label p))
```

Comment:
This is currently applying recursively for each call of `go`, resulting in extra labels showing up. It just needs to apply to `infer`/`check` below.

Reply:
I made these changes, it didn't fix all of the comments rhendic mentioned earlier. If you want to look for yourself you can see them here: https://github.com/purescript/purescript/pull/4416/commits/e136c2a6a79ec9c0fbe7de9d66ae8dc49f7553d6

I think the right decision is to not add the information there. I'd rather see this merged and then move to cleaning up the error messages in general. :D

Thanks for the input! :)